src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -107,24 +128,37 @@ public function createSshTunnel(array $dbConfig)
         return $dbConfig;
     }
 
-    abstract public function createConnection($params);
+    /**
+     * @param array $params
+     * @return \PDO|mixed
+     */
+    abstract public function createConnection(array $params);
 
+    /**
+     * @return void|mixed
+     */
     abstract public function testConnection();
 
     /**
      * @param array|null $tables - an optional array of tables with tableName and schema properties
      * @return mixed
      */
-    abstract public function getTables(array $tables = null);
+    abstract public function getTables(?array $tables = null): array;
 
-    abstract public function simpleQuery(array $table, array $columns = array());
+    abstract public function simpleQuery(array $table, array $columns = array()): string;
 
-    public function validateIncrementalFetching(array $table, string $columnName, int $limit = null)
+    /**
+     * @param array    $table
+     * @param string   $columnName
+     * @param int|null $limit
+     * @throws UserException
```

Comment:
I don't think we use `@throws`. We can discuss this... https://keboola.slack.com/archives/C9FEAHW7N/p1525259533000099